test/python/test_rticonnextdds_data_access.py

```diff
@@ -393,6 +403,54 @@ def test_shrink_sequence(self, test_output, test_input, test_dictionary):
     assert sample["my_point_array[0].y"] == 20 # Retains value
     assert sample["my_point_array[4].x"] == 14 # Retains value
 
+  def test_too_large_uint64_output(self, test_output):
+    with pytest.raises(rti.Error, match=r".*value of my_uint64 is too large.*") as execinfo:
+      test_output.instance.set_number("my_uint64", 9007199254740992)
+    with pytest.raises(rti.Error, match=r".*value of my_int64 is too large.*") as execinfo:
+      test_output.instance.set_number("my_int64", -9007199254740992)
+
+  def verify_large_integer(self, output, input, number):
+    with pytest.raises(rti.Error, match=r".*value of my_uint64 is too large.*") as execinfo:
+      output.instance.set_number("my_uint64", number)
+    with pytest.raises(rti.Error, match=r".*value of my_int64 is too large.*") as execinfo:
+      output.instance.set_number("my_int64", -number)
+
+    output.instance.set_dictionary({"my_uint64": number, "my_int64":-number})
+    sample = self.send_data(output, input)
+    dictionary = sample.get_dictionary()
+    assert dictionary["my_uint64"] == number
+    assert dictionary["my_int64"] == -number
+    with pytest.raises(rti.Error, match=r".*value of my_uint64 is too large.*") as execinfo:
+      sample.get_number("my_uint64")
+    with pytest.raises(rti.Error, match=r".*value of my_int64 is too large.*") as execinfo:
+      sample.get_number("my_int64")
+
+  def test_large_uint64(self, test_output, test_input):
+    max_int_get = 2**53 # 9007199254740992
+    max_int_set = 2**53 - 1
+
+    # largest integer allowed in set_number
+    test_output.instance.set_number("my_uint64", max_int_set)
+    test_output.instance.set_number("my_int64", -max_int_set)
+    sample = self.send_data(test_output, test_input)
+    assert sample.get_number("my_uint64") == max_int_set
+    assert sample.get_number("my_int64") == -max_int_set
+
+    # largest integer allowed in get_number; ok in set_dictionary
+    test_output.instance.set_dictionary({"my_uint64":max_int_get})
+    test_output.instance.set_dictionary({"my_int64":-max_int_get})
+    sample = self.send_data(test_output, test_input)
+    assert sample.get_number("my_uint64") == max_int_get
+    assert sample.get_number("my_int64") == -max_int_get
+
+    # too large for get_number, but ok in get_dictionary
+    self.verify_large_integer(test_output, test_input, max_int_get + 1)
+
+    # 9007199254740999 -> 9007199254741000
+    self.verify_large_integer(test_output, test_input, 9007199254740999)
+
+    # largest long long
```

Comment:
Do you think we should add a test which uses these same values but set/get_dictionary?